guide_drafts/trait_derives.md

```diff
@@ -93,96 +104,177 @@ fn main() {
+
+The implementation of `QueryableByName` will assume that columns with a given name will 
+have a certain type.
+Be careful, as the compiler will be unable to verify that the given column type is correct.
+If your query returns a column of an unexpected type,
+the result may have the wrong value, or return an error.
+
+In order to use `QueryableByName`,
+you are required to use one of the following annontations.
+If your struct references a single table,
+you may annontate the struct with `#[table_name="my_table"]`.
+Alternatively you may also annotate each field on your struct
+with `#[sql_type="ColumnTypeHere"]`.
+Both of these will be demonstrated in the following example.
 
-```rust
+### Example 
 
-// Implementing the Queryable trait by hand
+```rust
+// File: src/models.rs
 
-use diesel::query_source::Queryable;
-use diesel::backend::Backend;
-use diesel::types::{Integer, Text, HasSqlType, FromSqlRow};
+use diesel::types::Text;
+use schema::users; // Brings table! into scope
 
-pub struct User {
+#[derive(Debug, QueryableByName)]
+#[table_name="users"]
+pub struct UserEmail {
     pub id: i32,
+    pub email: String,
+}
+
+#[derive(Debug, QueryableByName)]
+pub struct UserName {
+    #[sql_type="Text"]
     pub first_name: String,
+    #[sql_type="Text"]
     pub last_name: String,
-    pub email: String,
 }
 
-impl<ST, DB> Queryable<ST, DB> for User where
-    DB: Backend + HasSqlType<ST>,
-    (i32, String, String, String): FromSqlRow<ST, DB> {
-    
-    type Row = (i32, String, String, String);
-
-    fn build(row: &Self::Row) -> Self {
-        User {
-            id: row.0,
-            first_name: row.1,
-            last_name: row.2,
-            email: row.3,
-        }
-    }
+#[derive(Insertable)]
+#[table_name="users"]
+pub struct NewUser<'a> {
+    pub first_name: &'a str,
+    pub last_name: &'a str,
+    pub email: &'a str,
+ }
+
+```
+
+```rust
+// File: src/main.rs
+
+use diesel::prelude::*;
+
+fn main() {
+  let new_user = NewUser { 
+      first_name: "Gordon", 
+      last_name: "Freeman", 
+      email: "gordon.freeman@blackmesa.co" 
+  };
+
+  diesel::insert_into(users::table)
+      .values(&new_user)
+      .execute(&db_connection)
+      .expect("Error inserting row into database");
+
+  let users_emails = sql_query("SELECT users.id, users.email FROM users ORDER BY id")
+    .load::<UserEmail>(&connection);
+  println!("{:?}", users_emails); 
+  // > User { id: 1, email: "gordon.freeman@blackmesa.co" }
+  
+  let users_name = sql_query("SELECT users.first_name, users.last_name FROM users ORDER BY id")
+    .load::<UserName>(&connection);
+  println!("{:?}", users_name); 
+  // > User { first_name: "Gordon", last_name: "freeman" }
 }
 ```
 
-The advantages of using the derive annotation should be clear here, as the more columns you have, the more code you would have to write.
+If we were to forget one of `QueryableByName`'s required struct annotations,
+we would see the following error.
```

Comment:
It's probably worth mentioning the error someone would see if they have `#[table_name]` but forgot `#[sql_type]` on a field that isn't meant to map to a column, or if there was a type mismatch.